apps/namada-interface/src/App/Staking/StakingOverview/AllValidatorsTable/AllValidatorsTable.tsx

```diff
@@ -0,0 +1,178 @@
+      } else {
+        triangle = " \u25bf"; // white down-pointing small triangle
+      }
+    }
+
+    return `${column}${triangle}`;
+  }
+
+  return {
+    rowRenderer: AllValidatorsRowRenderer,
+    callbacks: {
+      onClickValidator: navigateToValidatorDetails,
+    },
+    columns: [
+      {
+        uuid: "1",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.Validator),
+        width: "45%",
+        onClick: () => onColumnClick(AllValidatorsColumn.Validator),
+      },
+      {
+        uuid: "2",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.VotingPower),
+        width: "25%",
+        onClick: () => onColumnClick(AllValidatorsColumn.VotingPower),
+      },
+      {
+        uuid: "3",
+        columnLabel: getLabelWithTriangle(AllValidatorsColumn.Commission),
+        width: "30%",
+        onClick: () => onColumnClick(AllValidatorsColumn.Commission),
+      },
+    ],
+  };
+};
+
+const sortValidators = (sort: Sort, validators: Validator[]): Validator[] => {
+  const direction = sort.ascending ? 1 : -1;
+
+  const ascendingSortFn: (a: Validator, b: Validator) => number =
+    sort.column === AllValidatorsColumn.Validator ?
+      (a, b) => a.name.localeCompare(b.name) :
+    sort.column === AllValidatorsColumn.VotingPower ?
+      ((a, b) =>
+        a.votingPower === undefined || b.votingPower === undefined ? 0 :
+        a.votingPower.isLessThan(b.votingPower) ? -1 : 1) :
+    sort.column === AllValidatorsColumn.Commission ?
+      ((a, b) => a.commission.isLessThan(b.commission) ? -1 : 1) :
+    assertNever(sort.column);
+
+  const cloned = validators.slice();
+  cloned.sort((a, b) => direction * ascendingSortFn(a, b));
+
+  return cloned;
+}
+
+const filterValidators = (search: string, validators: Validator[]): Validator[] =>
+  validators.filter(v =>
+    search === ""
+      ? true
+      : v.name.toLowerCase().startsWith(search.toLowerCase())
+  );
+
+const selectSortedFilteredValidators = (sort: Sort, search: string) =>
+  (state: RootState): Validator[] => {
+    const validators = state.stakingAndGovernance.validators;
+
+    const sorted = sortValidators(sort, validators);
+    const sortedAndFiltered = filterValidators(search, sorted);
+
+    return sortedAndFiltered;
+  };
+
+enum AllValidatorsColumn {
+  Validator = "Validator",
+  VotingPower = "Voting power",
+  Commission = "Commission"
+};
+
+type Sort = {
+  column: AllValidatorsColumn;
+  ascending: boolean;
+};
+
+export const AllValidatorsTable: React.FC<{
+  navigateToValidatorDetails: (validatorId: string) => void;
+}> = ({
+  navigateToValidatorDetails,
+}) => {
+  const [search, setSearch] = useState("");
+
+  const [sort, setSort] = useState<Sort>({
+    column: AllValidatorsColumn.Validator,
+    ascending: true
+  });
+
+  const sortedFilteredValidators = useAppSelector(
+    selectSortedFilteredValidators(sort, search)
+  );
+
+  const handleColumnClick = (column: AllValidatorsColumn): void =>
```

Comment:
I think this will be redefined every time that search state changes? So maybe it's worth wrapping in useCallback?

Reply:
I was a bit unsure when writing this code originally. With `useCallback` I would define the dependency list as `[sort, search]`, or something like that? But then I guess it would still get redefined. I think there is something in redux about memoizing selectors, but I had planned to look into it sometime in the future (and probably there is a lot we could do to optimize the UI rendering). What do you think?

Reply:
But do you need `search` as a dependency? You are only using sort here.  Or am I missing something? :)

Reply:
The selector filters the validators using the `search` value by calling the `filterValidators` function. So it's doing both filtering by the search term, and sorting by the selected column.